File: src/main/java/mygame/editor/model/CircleModel.java
package mygame.editor.model;

import javafx.geometry.Point2D;
import javafx.scene.shape.Circle;
import mygame.Constants;

/**
 * Created by oleh on 3/21/17.
 */
public class CircleModel extends AbstractModel {

    Circle circle;
    Circle handler;

    public CircleModel(Point2D pos){
        super(Type.CIRCLE);
        circle = new Circle(pos.getX(),pos.getY(),5, Constants.GREEN.deriveColor(1,1,1,.25));
            circle.setStroke(Constants.GREEN);
            getChildren().add(circle);



    }

    public void updateRadius(Point2D radius){
        if(handler == null) {
            handler = getHandler(radius);
            handler.setUserData(circle);
            addControl(handler);
        }else{
            handler.setCenterY(radius.getY());
            handler.setCenterX(radius.getX());
            circle.setRadius(radius.distance(circle.getCenterX(),circle.getCenterY()));
        }


    }

}

File: src/main/java/mygame/editor/model/CircleModel.kt
package mygame.editor.model

import javafx.scene.shape.Circle
import mygame.Constants


class CircleModel(pos: Point) : AbstractModel(Type.CIRCLE) {

    var circle:Circle
    val center:Point = pos
    val radiusPoint:Point = pos.clone().add(10.0,0.0)

    init{
        circle =  Circle(pos.x,pos.y,center.distance(radiusPoint), Constants.GREEN.deriveColor(1.0,1.0,1.0,.25))
        circle.stroke = Constants.GREEN
        appendPoint(pos)
        appendPoint(radiusPoint)
        children.add(circle)
    }

    fun updateRadius(radius:Point){
        radiusPoint.set(radius)

        circle.radius = center.distance(radiusPoint)
    }

    override fun transform(trans: CustonAffine) {
        val newCenter = trans.transformPoint(center)
        val newRadius = trans.transformPoint(radiusPoint)
        update(newCenter,newRadius)

    }

    override fun update(vararg points:Point) {

        var center:Point = points[0]
        var radius:Point =  points[1]

        circle.radius = center.distance(radius)
        circle.centerX = center.x
        circle.centerY = center.y
    }
}

